### PR TITLE
Ethernet header v5

### PR DESCRIPTION
**Pull request by regit**
Update of #2677 fixing comments.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/262
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/44

### CODE REVIEW

**Comment by inliniac** [comment]
This comment is still valid AFAICS https://github.com/inliniac/suricata/pull/2677#issuecomment-300398215

**Comment by regit** [comment]
I think we should fix it in the way we are generating pseudo packet. For tunnel (PacketTunnelPktSetup), I think this is not really accurate to have the ethernet header. For defrag (PacketDefragPktSetup), we can use the parent ethernet header.

File: suricata.yaml.in [hunk 1]
```diff
@@ -158,6 +158,8 @@ outputs:
       #  pipelining:
       #    enabled: yes ## set enable to yes to enable query pipelining
       #    batch-size: 10 ## number of entry to keep in buffer
+      # Log ethernet header in events when available
+      #log-ethernet: yes
```
Comment (jasonish) [inline suricata.yaml.in]: I think I'd rather see this as just "ethernet" to be consistent with all the other options, ie: We have "tls", not "log-tls".

**Comment by jasonish** [comment]
> This comment is still valid AFAICS #2677 (comment)

I like the idea of adding the mac address to the flow so its available for packets that might not have them. But I'm not sure how keen I am on complex handling of the mac address like tracking a list of addresses as it changes over time. I think simple is better for a new feature like this. Either the first mac addresses seen, or overwrite on change, but don't react on it for now.

**Comment by regit** [comment]
It took me some time but I'm now convinced that the whole thing is not correct. Adding the Ethernet header on Flow is just an approximation. And we may be fooled by corner case like injection. And what should be the Ethernet address for the pseudo packets ?
If we look at an alert on streaming, the current code is use the Ethernet header of the packet triggering the ACK but the content triggering the content is in the other side. So a strict approach seems be to add the Ethernet information in the stream segment and use that in the logging. But we possibly have a set of Ethernet headers here if the alert is triggered on a set of segments. This case seems a bit unlikely but it would nice to address it.

File: suricata.yaml.in (hunk 1 above)
Comment (regit) [inline suricata.yaml.in]: @jasonish it is not at the same level (not under alert) so I was thinking about being more explicit.

**Comment by inliniac** [comment]
The segment idea comes with it's own issues. We match on reassembled data, so containing multiple segments. In theory each could have unique MACs. So I guess that only in extremely simple cases (e.g. matching on icmp id) we could be 100% correct about the MAC. Otherwise it would be a best guess.

**Comment by inliniac** [comment]
The problem is similar to https://redmine.openinfosecfoundation.org/issues/2057

**Pull request closed without merging**